### ISSUE TITLE
mypy should also give error for the type alias set multiple `ParamSpec`-based type parameters not just for the type alias set multiple `TypeVarTuple`-based type parameters

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20545]
*Memo:
- mypy test.py
- mypy 1.19.1
- Python 3.14.0

Defining and using the type alias set multiple [ParamSpec](https://docs.python.org/3/library/typing.html#typing.ParamSpec)-based type parameters, mypy doesn't give error while defining and using the type alias set multiple [TypeVarTuple](https://docs.python.org/3/library/typing.html#typing.TypeVarTuple)-based type parameters, mypy gives the error as shown below so mypy should also give error for the type alias set multiple `ParamSpec`-based type parameters:

### Multiple `ParamSpec` based:

```python
''' New syntax '''
type TA[**P1, **P2] = Callable[P1, None] | Callable[P2, None] # No error
''' New syntax '''

''' Old syntax '''
# from typing import ParamSpec, TypeAlias

# P1 = ParamSpec('P1')
# P2 = ParamSpec('P2')

# TA: TypeAlias = Callable[P1, None] | Callable[P2, None] # No error
''' Old syntax '''

v: TA[[int, str], [bool, None]] # No error
```

### Multiple `TypeVarTuple` based:

```python
''' New syntax '''
type TA[*Ts1, *Ts2] = tuple[*Ts1] | tuple[*Ts2] # No error
''' New syntax '''

''' Old syntax '''
# from typing import TypeAlias, TypeVarTuple

# Ts1 = TypeVarTuple('Ts1')
# Ts2 = TypeVarTuple('Ts2')

# TA: TypeAlias = tuple[*Ts1] | tuple[*Ts2] # No error
''' Old syntax '''

v: TA[*tuple[int, ...], *tuple[str, ...]] # Error
```

> error: More than one variadic Unpack in a type is not allowed

**Comment by a12k** [comment]
I think I'm lost here. The first example seems fine because TA is just a container for two separate Callables. v is either a function that takes (int, str) OR a function that takes (bool, None). The second example I would expect to error out because you're trying to unpack two variadic types into one list of type arguments, which is explicitly forbidden by a PEP, and because *tuple[int, ...] and *tuple[str, ...] are of each unknown lengths, so combined they are mathematically indeterminate.

**Comment by A5rocks** [comment]
Yeah, the issue for TypeVarTuple is that there's not a good syntax for it. With ParamSpec there is, so no error necessary.
